components/crud-web-apps/jupyter/backend/apps/common/yaml/spawner_ui_config.yaml

```diff
@@ -18,29 +18,106 @@ spawnerFormDefaults:
+    # By default, custom container Images are allowed
+    # Uncomment the following line to only enable standard container Images
+    readOnly: false
+  imageRStudio:
+    # The container Image for the user's RStudio Server
+    # If readonly, this value must be a member of the list below
+    value: public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/rstudio:master-28af7716
+    # The list of available standard container Images
+    options:
+      - public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/rstudio:master-28af7716
+    logo: |-
```

Comment:
@DavidSpek taking your comment https://github.com/kubeflow/kubeflow/pull/5646#issuecomment-801308346 here to have everything coordinated in this place

> @kimwnasptd Could we do some sort of combination of the things you describe and what is being done now? More specifically, allow the user to set the URLs for the icons in the config map. 

This is the initial idea I had https://github.com/kubeflow/kubeflow/pull/5646#issuecomment-797521345, although thinking about it more I believe it will be more involved to get it working. In this case we will need to first fetch the yaml [ and the request could fail ] and then inject the urls and use them.

> Due to the use of trademarks and the applicable guidelines I think it is import to make it very easy for people to remove the logos (preferably without needing to build their own image).

Could you elaborate a little bit more on this? How does the trademarks and applicable guidelines fit into the picture?

I agree that we should make it easy for people to swap the logos they prefer, but these users are the minority and most probably won't have a hard time to rebuild the image.

Reply:
This was very briefly discussed on above. However, the RStudio guidelines for using their trademark are very strict (Jupyter is the complete opposite and definitely not a problem, VSCode is still unknown). For RStudio, the only permitted use that could apply is: "Your project deploys RStudio products in unmodified form supplied by RStudio, PBC **for internal use only**;". In other words, EKF can probably not use the RStudio logo (you build your own images anyways, but still), and any vendor that offers Kubeflow-as-a-Service (StatCan, though it might pass as internal for now) or provides a distribution. These uses would all need written approval from RStudio to use their logo in the UI. And VSCode is still unknown at this point. 
This is the reason why I would like to have a solution that doesn't require building a new image, but rather just changing a configmap or something similar. This logo issue is something that will also be faced if the Spawner UI becomes more configurable in the future (the admin can create whatever groups of images he wants, and the logos will need to come from somewhere).

Reply:
@kimwnasptd I think I just thought of the solution for this. If we use links for the logos that direct to the SVGs hosted in this repository, we can swap the SVGs for something else if using the logos turns out to be a problem. I believe if this would the propagate to all running installations as well (if the cache of the original SVG is cleared). Does this make sense?

Reply:
From what I understand the issue is not how the icon will appear to the UI [ via the backend directly, fetched from somewhere public etc ] but the fact that we use this logo in the software. With that I mean that if someone would use the app he would end up seeing the logo, which could imply that this is endorsed from RStudio etc.

Also an issue with fetching resources in the UI directly from the public internet is that the app unable will not work in air-gaped environments, where the users would only have access to specific IPs.

Reply:
So judging from our current situation, where we are tied by time constraints and the trademark situation would need some communication and tip toeing into what it's acceptable I would propose the following solution to unblock ourselves:

1. Don't use the logos in this PR. We could use the [group options](https://v8.material.angular.io/components/select/overview#creating-groups-of-options) from Angular material, or just use buttons with text [ would prefer the first approach tbh ] to show the images for the different servers
2. Open an issue about it and involve @castrojo in this. And once we figure out what we are allowed to do with the rstudio trademark we can open a new PR and add this functionality. We've done it once here so this won't be difficult to do in the future

WDYT?

Reply:
Pulling it from the repo wouldn't solve your first point, but it does blur the distribution and way the logo is implemented in the software a bit. It could even directly link to the original source of the RStudio SVG. Indeed, for airgaped environment this might pose a slight problem. However, I think that people in that situation could build their own image (and might already be doing so as they would have a private registry as well). Also, I don't think the functionality of the JWA would actually break, the button would just be empty. 

I think the other problem with this is decision is who is responsible for the decision to include the logos or not and who would be responsible if RStudio isn't happy. If I'm the person responsible since it is my PR that adds their logo, I'm fine with that (and would prefer this actually). If I can be the person responsible for this, I'd go for adding the logos and remove them if there is a complaint. I think an argument can be made that the logo is being used according to the guidelines, if internal use is interpreted as "the kubeflow/wg-notebooks developers deploying unmodified RStudio binaries supplied by RStudio to test their functionality in Kubeflow during development".

Reply:
I only just saw your second comment. Personally, I would then be more in favor of using the button toggle that is there now, but using an SVG with text in it for VSCode and RStudio. This still makes it rather easy for users to customize if they want the actual logos (which they might be allowed to use in their situation). It also allows us to keep using the Jupyter logo (as we can definitely use that one), and works together with https://github.com/kubeflow/kubeflow/pull/5646#pullrequestreview-614539090 as well (assuming I get this implemented quickly).

Reply:
Let's go with using the svgs. We'll use links to fetch them from where they publicly host them. Distributions could make a small change and add their svgs to the app and simply change these links, which is something that we were OK with from the beginning. Lets just add a small section for this in the app's README.

Regarding the responsibility part, I'm not a layer on this so not an expert but we are a team and all of us move forward together. We are not going to point fingers to people if things go south. And also since I'm a reviewer and agreed to move forward with this so I'm also responsible.

But if indeed people complain about the usage then we will sincerely apologize and just change this to something like we discussed above.